lib/node_modules/@stdlib/_tools/changelog/generate/lib/main.js

```diff
@@ -0,0 +1,388 @@
+* // returns '...'
+*/
+function generate( pkg, releaseType ) {
+	var isNamespacePkg;
+	var releaseCommits;
+	var newestRelease;
+	var bySubpackage;
+	var nextVersion;
+	var standalone;
+	var unreleased;
+	var releases;
+	var pkgNames;
+	var commits;
+	var version;
+	var summary;
+	var date;
+	var name;
+	var out;
+	var i;
+	var j;
+
+	if ( !isString( pkg ) ) {
+		throw new TypeError( format( 'invalid argument. Must provide a string. Value: `%s`.', pkg ) );
+	}
+	if ( pkg === '' ) {
+		// Case: root package
+		isNamespacePkg = true;
+		standalone = '@stdlib/stdlib';
+		releases = objectEntries( npmReleases( standalone ) );
+	} else {
+		// Case: all other packages
+		isNamespacePkg = contains( STDLIB_NAMESPACE_PKGS, '@stdlib/' + pkg );
+		standalone = '@stdlib/' + replace( pkg, '/', '-' );
+		releases = objectEntries( npmReleases( standalone ) );
+	}
+
+	out = '# CHANGELOG\n\n';
+	out += '> Package changelog.\n\n';
+
+	commits = parseCommits( pkg );
+	if ( !commits || commits.length === 0 ) {
+		throw new Error( format( 'invalid argument. Unable to parse commits for package: `%s`.', pkg ) );
+	}
+
+	// Filter out all commits which only affect internal tooling:
+	commits = filter( commits, isNotInternalTooling );
+
+	// Group commits by release:
+	commits = groupBy( commits, indicator );
+
+	if ( releaseType ) {
+		newestRelease = releases[ releases.length-1 ][ 0 ];
+		nextVersion = semver.inc( newestRelease, releaseType );
+		if ( isNull( nextVersion ) ) {
+			throw new TypeError( format( 'invalid argument. Unrecognized release type: `%s`.', releaseType ) );
+		}
+	}
+
+	if ( isNamespacePkg ) {
+		out += releaseSectionStart( nextVersion );
+		out += '## ' + ( nextVersion || 'Unreleased' ) + ' (' + formatDate() + ')\n\n';
+		bySubpackage = groupBySubPackage( commits.unreleased, pkg );
+		pkgNames = objectKeys( bySubpackage ).sort();
+		out += sectionStart( 'packages' );
+		out += heading( 'Packages', 3 );
+		for ( i = 0; i < pkgNames.length; i++ ) {
+			name = pkgNames[ i ];
+			unreleased = releaseSummary( bySubpackage[ name ], true, true );
+			if ( unreleased ) {
+				out += packageSummaryWrapper( pkg, '', name, unreleased );
+			}
+		}
+		out += sectionEnd( 'packages' );
+		out += breakingChanges( commits.unreleased );
+		out += closedIssues( commits.unreleased );
+		out += formatContributors( commits.unreleased );
+		out += formatCommits( commits.unreleased );
+		out += sectionEnd( 'release' );
+	} else {
+		unreleased = releaseSummary( commits.unreleased );
+		if ( unreleased ) {
+			out += releaseSectionStart( nextVersion );
+			out += '## ' + ( nextVersion || 'Unreleased' ) + ' (' + formatDate() + ')\n\n';
+			out += unreleased;
+			out += sectionEnd( 'release' );
+		}
+	}
+	if ( isNamespacePkg ) {
+		for ( i = releases.length-1; i >= 0; i-- ) {
+			version = releases[ i ][ 0 ];
+			releaseCommits = commits[ version ];
+			if ( !releaseCommits ) {
+				continue;
+			}
+			date = new Date( releases[ i ][ 1 ] ).toISOString().slice( 0, 10 );
```

Comment:
I believe you should be able to repurpose `formatDate` above, with possibly a small tweak.

Reply:
Addressed.